frontend/javascripts/admin/statistic/time_tracking_overview.tsx

```diff
@@ -0,0 +1,261 @@
+import { DownloadOutlined, FilterOutlined } from "@ant-design/icons";
+import saveAs from "file-saver";
+import { formatMilliseconds } from "libs/format_utils";
+import { Link } from "react-router-dom";
+import ProjectAndAnnotationTypeDropdown from "./project_and_annotation_type_dropdown";
+import { isUserAdminOrTeamManager } from "libs/utils";
+import messages from "messages";
+import Toast from "libs/toast";
+
+export enum AnnotationTypeFilterEnum {
+  ONLY_ANNOTATIONS_KEY = "Explorational",
+  ONLY_TASKS_KEY = "Task",
+  TASKS_AND_ANNOTATIONS_KEY = "Task,Explorational",
+}
+const { Column } = Table;
+const { RangePicker } = DatePicker;
+
+const TIMETRACKING_CSV_HEADER = ["userId,userFirstName,userLastName,timeTrackedInSeconds"];
+
+type TimeEntry = {
+  user: {
+    id: string;
+    firstName: string;
+    lastName: string;
+    email: string;
+  };
+  timeMillis: number;
+};
+
+function TimeTrackingOverview() {
+  const getTimeEntryUrl = (
+    startMs: number,
+    endMs: number,
+    teamIds: string[],
+    projectIds: string[],
+  ) => {
+    // Omit project parameter in request if annotation data is requested
+    const projectsParam = projectIds.length > 0 ? `&projectIds=${projectIds.join(",")}` : "";
+    return `api/time/summed/userList?start=${startMs}&end=${endMs}&annotationTypes=${selectedTypes}&teamIds=${teamIds.join(
+      ",",
+    )}${projectsParam}`;
+  };
+
+  const currentTime = dayjs();
+  const [startDate, setStartDate] = useState(currentTime.startOf("month"));
+  const [endDate, setEndeDate] = useState(currentTime);
+  const [isFetching, setIsFetching] = useState(false);
+  const mayUserAccessView = useFetch(
+    async () => {
+      const activeUser = await getActiveUser();
+      return isUserAdminOrTeamManager(activeUser);
+    },
+    false,
+    [],
+  );
+  const [allTeams, allProjects, allTimeEntries] = useFetch(
+    async () => {
+      if (!mayUserAccessView) return [[], [], []];
+      setIsFetching(true);
+      const [allTeams, allProjects] = await Promise.all([getTeams(), getProjects()]);
+      const timeEntriesURL = getTimeEntryUrl(
+        startDate.valueOf(),
+        endDate.valueOf(),
+        allTeams.map((team) => team.id),
+        allProjects.map((projects) => projects.id),
+      );
+      const allTimeEntries: TimeEntry[] = await Request.receiveJSON(timeEntriesURL);
+      setIsFetching(false);
+      return [allTeams, allProjects, allTimeEntries];
+    },
+    [[], [], []],
+    [mayUserAccessView],
+  );
+
+  const [selectedProjectIds, setSelectedProjectIds] = useState(Array<string>);
+  const [selectedTypes, setSelectedTypes] = useState(
+    AnnotationTypeFilterEnum.TASKS_AND_ANNOTATIONS_KEY,
+  );
+  const [selectedTeams, setSelectedTeams] = useState(allTeams.map((team) => team.id));
+  const [projectOrTypeQueryParam, setProjectOrTypeQueryParam] = useState("");
+  useEffect(() => {
+    if (selectedProjectIds.length > 0 && selectedProjectIds.length < allProjects.length) {
```

Comment:
never, but they can be the same length if all projects are selected. In that case I want to use the `project` type instead of listing all project ids as the query param

Reply:
ahh, I see :+1: